cmd/thv-operator/controllers/mcpserver_controller.go

```diff
@@ -699,6 +699,26 @@ func (r *MCPServerReconciler) deploymentForMCPServer(ctx context.Context, m *mcp
 		configMapName := fmt.Sprintf("%s-runconfig", m.Name)
 		configMapRef := fmt.Sprintf("%s/%s", m.Namespace, configMapName)
 		args = append(args, fmt.Sprintf("--from-configmap=%s", configMapRef))
+
+		// Also add pod template patch for secrets and service account (same as regular flags approach)
+		serviceAccount := m.Spec.ServiceAccount
```

Comment:
Interested where this is coming from? Do we know why we need to add this? As far as I'm aware the service account stuff worked with the configmap?